### DISCUSSION TITLE
Fatal error: main() [[function.require](http://ccbe.link/admin/function.require)]: Failed opening required './includes/load-yourls.php'

### DISCUSSION THREAD

**Discussion by matbe81** [YOURLS/YOURLS#3510] [General]
I get suddently the following message when trying to access [http://ccbe.link/admin](http://ccbe.link/admin) while everything was working for months without any changes.

Warning: main(./includes/load-yourls.php) [[function.main](http://ccbe.link/admin/function.main)]: failed to open stream: No such file or directory in /home/filigran/ccbe.link/admin/index.php on line 3

Fatal error: main() [[function.require](http://ccbe.link/admin/function.require)]: Failed opening required './includes/load-yourls.php' (include_path='.:/usr/local/lib/php') in /home/filigran/ccbe.link/admin/index.php on line 3

**Comment by dgw** [comment]
Something clearly changed, because that file is part of the YOURLS install package. Your web host may be having storage problems and the `load-yourls.php` file got lost.

Based on #3509 I suggest you reinstall 1.7.2, then upgrade YOURLS as far as you can. [1.7.2 will celebrate](https://github.com/YOURLS/YOURLS/releases/tag/1.7.2) its sixth birthday in a few months.

**Comment by matbe81** [comment]
Thank you. But I got exactly the same error message when overwriting with 1.7.2 original package.

**Comment by matbe81** [comment]
I Found the cause: Running in PHP 7.0. Working back in PHP 5.4.
I have not upgraded to 1.9.1 and set PHP version to 5.6 but got this error message trying to access /admin:
Composer detected issues in your platform: Your Composer dependencies require a PHP version ">= 7.4.0".

**Reply by dgw** [reply]
Upgrading "as far as you can" means, subject to the site hosting environment. YOURLS dropped support for PHP5 in version 1.8, and now requires PHP 7.4+.

The last PHP5 branch (5.6) stopped getting updates 4 years ago. If your hosting provider does not support even PHP 7.4, please ask their tech support department what their plans are. They should be providing PHP 8.x by now.

**Comment by matbe81** [comment]
The hosting service is OVH, a well known French company. They support those PHP versions:
5.4, 5.5, 5.6 and 7.0
7.1, 7.2 and 7.3
7.4, 8.0 and 8.1 (bêta)

The issue is even if I choose a more recent version like 7.4, I got this error message:

Not Implemented
GET not supported for current URL.
Additionally, a 501 Not Implemented error was encountered while trying to use an ErrorDocument to handle the request.

**Reply by ozh** [reply]
Generally, when the problem is along the lines of "everything was running smoothly and I didn't change anything and now it's broken", 

1. we're not much of an help since the problem is specific to the mentioned install
2. something **has** changed

**Reply by matbe81** [reply]
I generally agree with you. But "I didn't change anything" doesn't means "something else changed". Like hosting environment for example. But this doesn't explain the latest error message I got while trying to upgrade to 7.4.

**Reply by dgw** [reply]
There are a number of threads on OVH's forum (in French) about the 501 Not Implemented error in PHP 7.4. Skimming through those suggests that there might be quirks to how OVH PHP version selection works. I'd encourage you to consult OVH's documentation.

**Reply by matbe81** [reply]
Indeed. Thank you !

**Reply by ozh** [reply]
Unsolicited advice but I'll give it anyway: avoid OVH, seriously. Their hosting is ok, their features are just below ok, and their user dashboard / control panel, omg the absolute horror. When I was freelancing I used to turn down all projects hosted on OVH, for real.
